internal/config/merged_config.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"errors"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/spf13/viper"
+)
+
+// Reads configurations from all specified files using read(),
+// and then merges them into a single viper.Viper instance.
+func readMerged(files ...string) (View, error) {
+	if len(files) == 0 {
+		return nil, errors.New("no input files specified")
+	}
+
+	w := new(wrapperView)
+	layers := make([]*viper.Viper, len(files))
+
+	queue := make(chan fsnotify.Event, 1)
+	onFileChange := func(e fsnotify.Event) {
+		select {
+		case queue <- e:
+		default:
+		}
+	}
+
+	// read files into layers and watch for changes
+	for i, f := range files {
+		l, err := read(f, onFileChange)
+		if err != nil {
+			return nil, err
+		}
+		layers[i] = l
+	}
+
+	w.cfg = merge(layers...)
+
+	// re-merge layers upon changes in files
+	go func() {
+		for range queue {
+			w.cfg = merge(layers...)
+		}
+	}()
+
+	return w, nil
+}
+
+func merge(layers ...*viper.Viper) *viper.Viper {
+	cfg := viper.New()
+	for _, l := range layers {
+		m := l.AllSettings()
+		cfg.MergeConfigMap(m)
+	}
+	return cfg
+}
+
+// Wrapper struct that implements View interface
+// and delegates to other viper.Viper instance
+type wrapperView struct {
```

Comment:
Why is this necessary if you have the merge() function?

Reply:
I didn't find a function to replace all contents of a Viper instance. That means that if we use the same instance upon layers' reloads, then that instance might become polluted with some old data from previous reads.